src/ray/gcs/gcs_server/gcs_actor_manager.cc

```diff
@@ -1453,10 +1453,15 @@ void GcsActorManager::RestartActor(const ActorID &actor_id,
       << ", death context type = " << GetActorDeathCauseString(death_cause)
       << ", remaining_restarts = " << remaining_restarts;
 
-  if (remaining_restarts != 0) {
+  if (remaining_restarts != 0 ||
+      (need_reschedule && max_restarts > 0 && mutable_actor_table_data->preempted())) {
```

Comment:
Even when remaining_restarts == 0, we should still restart the actor if it was preempted.